### PR TITLE
Loaders and Tweaks

### PR DESCRIPTION
**Pull request by Camwyn**
**Fixed** incorrect calcs and rounding errors.
**Fixed** PHP notice about class_name at line 40 in mini-cart.php
**Fixed** PHP fatal boolean problem at line 48 in mini-cart.php
**Fixed** When loading qty from cart to tickets block, a sold out event gets that qty loaded and lets you set quantity (if you had a ticket for that sold out event in your cart before it was sold out)
**Fixed** Text at bottom of AR Page shows "There are X other tickets" message but the X is way wrong and varied in how wrong it is.
**Fixed** The message at the bottom of the page should not show if the number is 0
**Fixed** The number does not update if you remove items from the modal cart.

**Changed** optout text.

**Feature** Adds loader in footer of all event pages but only activates it (via js) on ticket (single event) and AR page.

🎥 http://p.tri.be/djESMv

### CODE REVIEW

File: src/Tribe/Tickets_View.php [hunk 1]
```diff
@@ -386,6 +388,49 @@ public function intercept_content( $content = '' ) {
 		return $content;
 	}
 
+	/**
+	 * Add spinner to event pages.
+	 *
+	 * @return void
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets_View.php]:
```suggestion
	 * @since TBD
```

File: src/views/registration-js/content.php [hunk 2]
```diff
@@ -14,7 +14,15 @@
...
 $template = tribe( 'tickets.editor.template' );
 
-$provider             = tribe_get_request_var( 'provider' );
+$provider = $this->get( 'provider' ) ?: tribe_get_request_var( 'provider' );
+
+if ( empty( $provider ) ) {
+	$provider_name     = Tribe__Tickets__Tickets::get_event_ticket_provider( array_key_first( $events ) );
+	$provider = $provider_name::ATTENDEE_OBJECT;
+}
+
```
Comment (sc0ttkclark) [inline src/views/registration-js/content.php]:
```suggestion
```

File: src/views/modal/registration-js.php [hunk 3]
```diff
@@ -62,7 +62,12 @@ class="tribe-tickets__item__attendee__fields__form <?php echo sanitize_html_clas
 					</h3>
 				</div>
 		<?php endforeach; ?>
-		<p class="tribe-tickets-notice tribe-tickets-notice--non-ar">
+		<p
+			class="tribe-tickets-notice tribe-tickets-notice--non-ar"
+			<?php if ( empty( $non_meta_count ) ) : ?>
+				style="display: none;"
```
Comment (sc0ttkclark) [inline src/views/modal/registration-js.php]: I thought inline `style="display:none"` was not recommended and we should be using a class instead, is that right?

File: src/views/registration-js/content.php [hunk 4]
```diff
@@ -125,7 +135,12 @@ class="tribe-tickets__item__attendee__fields__form <?php echo sanitize_html_clas
 		</div>
 	</div>
 	<div class="tribe-tickets__registration__footer">
-		<p class="tribe-tickets-notice tribe-tickets-notice--non-ar">
+		<p
+			class="tribe-tickets-notice tribe-tickets-notice--non-ar"
+			<?php if ( empty( $non_meta_count ) ) : ?>
+				style="display: none;"
```
Comment (sc0ttkclark) [inline src/views/registration-js/content.php]: I thought inline `style="display:none"` was not recommended and we should be using a class instead, is that right?

File: src/views/modal/cart.php [hunk 5]
```diff
@@ -45,5 +45,12 @@
 			<?php $template_obj->template( 'blocks/tickets/item', [ 'ticket' => $ticket, 'key' => $key, 'is_modal' => true, 'currency_symbol' => $currency_symbol ] ); ?>
 		<?php endforeach; ?>
 	<?php endif; ?>
+
+
+	<?php
+	$text    = apply_filters( 'tribe_tickets_loader_text', 'One Moment...' );
```
Comment (sc0ttkclark) [inline src/views/modal/cart.php]: Needs phpdoc

File: src/views/modal/cart.php [hunk 6]
```diff
@@ -45,5 +45,12 @@
...
+	$classes = apply_filters( 'tribe_tickets_modal_loader_classes', [ 'tribe-loader__modal' ] );
```
Comment (sc0ttkclark) [inline src/views/modal/cart.php]: Needs phpdoc

File: src/views/components/loader.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * See more documentation about our views templating system.
+ *
+ * @link {INSERT_ARTCILE_LINK_HERE}
+ *
+ * @version TBD
+ *
+ */
+if ( empty( $text ) ) {
+	$text = $this->get( 'text' ) ?: 'Loading...';
```
Comment (sc0ttkclark) [inline src/views/components/loader.php]:
```suggestion
	$text = $this->get( 'text' ) ?: __( 'Loading...', 'event-tickets' );
```

Comment (sc0ttkclark) [reply src/views/components/loader.php]: Also I keep confusing ?: and ?? and it's PHP version compatibility :) had to double check again!

Comment (Camwyn) [reply src/views/components/loader.php]: Elvis!

File: src/views/modal/registration-js.php (hunk 3 above)
Comment (Camwyn) [reply src/views/modal/registration-js.php]: We're using hide/show on this - which adds/removes the inline style. I didn't want to make the notices hidden by default, and some of the classes for that are harder to override.

That said, open to suggestions.

File: src/views/registration-js/content.php (hunk 4 above)
Comment (Camwyn) [reply src/views/registration-js/content.php]: See comment above: https://github.com/moderntribe/event-tickets/pull/1382/files#r330655751

File: src/views/modal/cart.php [hunk 8]
```diff
@@ -45,5 +45,26 @@
...
+
+	<?php
+	/**
+	 * Allows filtering of text used in the loader
+	 *
+	 * @since  TBD
+	 *
+	 * @param  string $value     The value that will be filtered.
+	 */
+	$text    = apply_filters( 'tribe_tickets_loader_text', 'One Moment...' );
```
Comment (sc0ttkclark) [inline src/views/modal/cart.php]:
```suggestion
	$text    = apply_filters( 'tribe_tickets_loader_text', __( 'One Moment...', 'event-tickets' ) );
```

File: src/views/blocks/tickets/extra-available-quantity.php [hunk 9]
```diff
@@ -18,5 +18,5 @@
 
 $ticket = $this->get( 'ticket' );
 ?>
-<span class="tribe-tickets__item__extra__available_quantity"><?php echo esc_html( $ticket->available() ); ?></span>
+<span class="tribe-tickets__item__extra__available_quantity"><?php echo esc_html( $ticket->available() ); ?><?php echo $ticket->global_stock_mode() ?></span>
```
Comment (sc0ttkclark) [inline src/views/blocks/tickets/extra-available-quantity.php]:
```suggestion
<span class="tribe-tickets__item__extra__available_quantity"><?php echo esc_html( $ticket->available() ); ?><?php echo $ticket->global_stock_mode(); ?></span>
```

File: src/Tribe/Tickets_View.php [hunk 10]
```diff
@@ -386,6 +388,63 @@ public function intercept_content( $content = '' ) {
...
+		ob_start();
+
+		/**
+		 * Allows filtering of text used in the loader
+		 *
+		 * @since  TBD
+		 *
+		 * @param  string $value     The value that will be filtered.
+		 */
+		$text = apply_filters( 'tribe_tickets_loader_text', 'Loading...' );
```
Comment (sc0ttkclark) [inline src/Tribe/Tickets_View.php]:
```suggestion
		$text = apply_filters( 'tribe_tickets_loader_text', __( 'Loading...', 'event-tickets' ) );
```

**Pull request merged**